### PR TITLE
Added real names to the activity tab

### PR DESCRIPTION
**Pull request by shishiro26**
**NOTE:** Please review the pull request process before opening your first PR: [WikiEduDashboard Contribution Guide](https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/master/CONTRIBUTING.md#pull-request-process)

## Summary of Changes
This PR enhances the Activity tab by displaying users' real names alongside their usernames, similar to the existing layout in the Students tab.

## Screenshots
**Before:**
image

**After:**
- **User View:**  
    Screenshot from 2024-11-14 22-32-32

- **Admin/Instructor View:**  
  Screenshot from 2024-11-14 22-32-41

## Open Questions and Concerns
I apologize for the mistake on my end; I accidentally deleted my comment in the issue where I had mentioned my interest in working on it.

**Related Issue:** #6034

### CODE REVIEW

File: app/assets/javascripts/components/revisions/revision.jsx [hunk 1]
```diff
@@ -3,13 +3,15 @@ import PropTypes from 'prop-types';
...
+import { trunc } from '../../utils/strings.js';
 
-const Revision = ({ revision, index, wikidataLabel, course, setSelectedIndex, lastIndex, selectedIndex }) => {
+const Revision = ({ revision, index, wikidataLabel, course, setSelectedIndex, lastIndex, selectedIndex, student }) => {
   const ratingClass = `rating ${revision.rating}`;
   const ratingMobileClass = `${ratingClass} tablet-only`;
   const formattedTitle = CourseUtils.formattedArticleTitle({ title: revision.title, project: revision.wiki.project, language: revision.wiki.language }, course.home_wiki, wikidataLabel);
   const subtitle = wikidataLabel ? `(${CourseUtils.removeNamespace(revision.title)})` : '';
   const isWikipedia = revision.wiki.project === 'wikipedia';
+  const showRealName = student.first_name && student.last_name;
```
Comment (ragesoss) [inline app/assets/javascripts/components/revisions/revision.jsx:14]: Why is this based on first_name and last_name, but real_name is the one that is actually used?

**Comment by ragesoss** [comment]
Cool! Can you include a screenshot of the instructor/admin view with more than one student, where one has a real name and another does not?

File: app/assets/javascripts/components/revisions/revision.jsx (hunk 1 above)
Comment (shishiro26) [reply app/assets/javascripts/components/revisions/revision.jsx:14]: You're right, that's my mistake. I'll change it to the real_name

**Comment by ragesoss** [comment]
I meant a screenshot that has a mix of real names and no real names.

**Comment by shishiro26** [comment]
Adding a real name is mandatory; without it, a user cannot join any course. Therefore, having a mix of real names and non-real names, as shown in the above case, is not possible .. 
  image

**Comment by shishiro26** [comment]
Adding a real name is mandatory; without it, a user cannot join any course. Therefore, having a mix of real names and non-real names, as shown in the above case, is not possible

image

**Comment by ragesoss** [comment]
As an instructor, you can add a user via their username from the Students tab. Adding a student this way does not require (or permit) a real name.

**Comment by shishiro26** [comment]
Yupp got it 
image

**Pull request merged**